### PR TITLE
cpu, cc2538: enhance and cleanup periph timer

### PR DESCRIPTION
**Pull request by smlng**
follow up on discussion in #7335, this makes enhancements and cleanup of low-level timer for cc2538. Fixes a problem, when timer are configured out of order, board configurations are adapted accordingly.

### CODE REVIEW

**Comment by smlng** [comment]
will adapt optimizations from #7335

**Comment by smlng** [comment]
@A-Paul and @haukepetersen can you have a look at this one, its basically a copy of the reworked periph timer for the cc26x0.

File: cpu/cc2538/periph/timer.c [hunk 1]
```diff
@@ -27,13 +27,13 @@
 #include "periph/timer.h"
 #include "periph_conf.h"
 
-#define ENABLE_DEBUG (0)
+#define ENABLE_DEBUG (1)
```
Comment (A-Paul) [inline cpu/cc2538/periph/timer.c]: Really?

Comment (smlng) [reply cpu/cc2538/periph/timer.c]: oops

**Pull request merged**